File: tests/fixtures/sample.rb
require 'json'
require_relative './helper'

class UserService
  def initialize(db)
    @db = db
  end

  def get_user(id)
    result = @db.query(id)
    format_user(result)
  end

  def delete_user(id)
    @db.execute("DELETE FROM users WHERE id = ?", id)
  end
end

def format_user(data)
  data.to_s.upcase
end

File: tests/fixtures/sample.kt
import java.util.Date

class UserService(private val db: Database) {
    fun getUser(id: Int): User {
        val result = db.query(id)
        return formatUser(result)
    }

    fun deleteUser(id: Int) {
        db.execute("DELETE FROM users WHERE id = ?", id)
    }
}

fun formatUser(data: Any): User {
    return User(data.toString())
}
